OpenRA.Mods.Common/Traits/Health.cs

```diff
@@ -125,6 +123,15 @@ void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newO
 			notifyKilledPlayer = newOwner.PlayerActor.TraitsImplementing<INotifyKilled>().ToArray();
 		}
 
+		public void ModifyHP(int addHP, int addMaxHP)
+		{
+			if (IsDead)
+				return;
+
+			hp += addHP;
+			MaxHP += addMaxHP;
```

Comment:
We probably should make sure that `hp` isn't increased above the (new) `MaxHP` value.